task/control-plane-agent/src/mgs_common.rs

```diff
@@ -23,6 +23,7 @@ task_slot!(PACKRAT, packrat);
 /// Provider of MGS handler logic common to all targets (gimlet, sidecar, psc).
 pub(crate) struct MgsCommon {
     reset_requested: bool,
+    reset_component_requested: SpComponent,
```

Comment:
What does it mean to set this when `reset_requested` is `false`?

Should `reset_requested` become an `Option<SpComponent>`?